realm/src/main/java/io/realm/Realm.java

```diff
@@ -194,10 +194,14 @@ public static Realm getInstance(Context context) {
      * model classes or version has has changed so a migration is required.
      */
     public static Realm getDefaultInstance() {
-        if (defaultConfiguration == null) {
+        final RealmConfiguration configuration;
+        synchronized (Realm.class) {
+            configuration = defaultConfiguration;
+        }
```

Comment:
To be honest, I don't see this scenario happen either. Currently, in the worst case, we end up calling create twice (which is synchronized, so second time would hit the cache), but this is extremely unlikely. So in this case I don't think it is worth adding the code complexity.